modules/mesh-layers/src/simple-mesh-layer/simple-mesh-layer.js

```diff
@@ -265,12 +237,11 @@ export default class SimpleMeshLayer extends Layer {
     }
 
     if (src) {
-      getTexture(gl, src).then(texture => {
-        this.setState({texture});
-        if (this.state.model) {
-          this.state.model.setUniforms({sampler: this.state.texture, hasTexture: 1});
-        }
-      });
+      const texture = getTextureFromData(gl, src);
+      this.setState({texture});
+      if (this.state.model) {
+        this.state.model.setUniforms({sampler: this.state.texture, hasTexture: 1});
+      }
```

Comment:
Nit: More elegant to just have the if statement calculate the texture and the flag and have a single set of calls to setState and setUniforms at the end.